scheduler/scheduler.go

```diff
@@ -187,6 +215,9 @@ func BuildDependencyGraph(c client.Interface, sel labels.Selector) (DependencyGr
 			depGraph[d.Child].Requires, depGraph[parent])
 		depGraph[parent].RequiredBy = append(
 			depGraph[parent].RequiredBy, depGraph[child])
+		if err := depGraph[parent].updateMeta(d.Meta); err != nil {
```

Comment:
And this code also shows, that we keep exact relation '1 scheduled resource <-> 1 resource'. In this case I'd made meta a part of resource, not scheduled resource. And we could keep resource interface (Create, Status) untouched.
But in such case success factor would be the same for all dependencies to the given RC.